arg_scanner/lib/arg_scanner/type_tracker.rb

```diff
@@ -13,7 +56,9 @@ class TypeTracker
     def initialize
       @cache = Set.new
       @socket = TCPSocket.new('127.0.0.1', 7777)
-
+      @mutex = Mutex.new
```

Comment:
I'd suggest instantiating it only on some condition (debugging env variable, etc)

Reply:
sure

Reply:
btw I think `@perf_mon = TTPM.new if @enable_debug` will do the same :)
But we'll do it LATER